File: composeApp/src/iosMain/kotlin/com/spoonofcode/dojopro/core/ui/utils/FirebasePushService.swift
package com.spoonofcode.dojopro.core.ui.utils

import Foundation
import FirebaseMessaging

@objc class FirebasePushServiceObjC: NSObject {
    @objc static func getToken(completion: @escaping (String?) -> Void) {
        Messaging.messaging().token { token, error in
            if let error = error {
                print("Error getting FCM token: \(error)")
                completion(nil)
            } else {
                completion(token)
            }
        }
    }

    @objc static func subscribe(topic: String) {
        Messaging.messaging().subscribe(toTopic: topic)
    }

    @objc static func unsubscribe(topic: String) {
        Messaging.messaging().unsubscribe(fromTopic: topic)
    }
}

File: composeApp/src/iosMain/kotlin/com/spoonofcode/dojopro/core/ui/utils/FirebasePushService.kt
package com.spoonofcode.dojopro.core.ui.utils

actual class FirebasePushService {
//    actual suspend fun getMessageToken(): String? = suspendCancellableCoroutine { cont ->
//        FirebasePushServiceObjC.getToken { token ->
//            cont.resume(token)
//        }
//    }
//
//    actual fun subscribeToTopic(topic: String) {
//        FirebasePushServiceObjC.subscribeWithTopic(topic)
//    }
//
//    actual fun unsubscribeFromTopic(topic: String) {
//        FirebasePushServiceObjC.unsubscribeWithTopic(topic)
//    }
    actual fun getMessageToken(): String {
        TODO("Not yet implemented")
    }

    actual suspend fun subscribeToTopic(topic: String) {
        TODO("Not yet implemented")
    }

    actual suspend fun unsubscribeFromTopic(topic: String) {
        TODO("Not yet implemented")
    }
}